tests/unit-core/test-promise.c

```diff
@@ -114,61 +113,79 @@ register_js_function (const char *name_p, /**< name of the function */
   jerry_release_value (result_val);
 } /* register_js_function */
 
+/**
+ * Checks whether global object has promise.
+ */
+static bool
+promise_is_available (void)
+{
+  jerry_value_t global_obj_val = jerry_get_global_object ();
+  jerry_value_t prop_name = jerry_create_string ((const jerry_char_t *) "Promise");
+
+  jerry_value_t prop_value = jerry_has_property (global_obj_val, prop_name);
+  bool has_prop = jerry_get_boolean_value (prop_value);
+
+  jerry_release_value (global_obj_val);
+  jerry_release_value (prop_name);
+  jerry_release_value (prop_value);
+
+  return has_prop;
+} /* promise_is_available */
+
 int
 main (void)
 {
   jerry_init (JERRY_INIT_EMPTY);
 
-  register_js_function ("create_promise1", create_promise1_handler);
-  register_js_function ("create_promise2", create_promise2_handler);
-  register_js_function ("assert", assert_handler);
-
-  jerry_value_t parsed_code_val = jerry_parse ((jerry_char_t *) test_source, strlen (test_source), false);
-  TEST_ASSERT (!jerry_value_has_error_flag (parsed_code_val));
-
-  jerry_value_t res = jerry_run (parsed_code_val);
-  TEST_ASSERT (!jerry_value_has_error_flag (res));
+  if (promise_is_available ())
+  {
+    register_js_function ("create_promise1", create_promise1_handler);
+    register_js_function ("create_promise2", create_promise2_handler);
+    register_js_function ("assert", assert_handler);
 
-  jerry_release_value (res);
-  jerry_release_value (parsed_code_val);
+    jerry_value_t parsed_code_val = jerry_parse ((jerry_char_t *) test_source, strlen (test_source), false);
+    TEST_ASSERT (!jerry_value_has_error_flag (parsed_code_val));
 
-  /* Test jerry_create_promise and jerry_value_is_promise. */
-  TEST_ASSERT (jerry_value_is_promise (my_promise1));
-  TEST_ASSERT (jerry_value_is_promise (my_promise2));
+    jerry_value_t res = jerry_run (parsed_code_val);
+    TEST_ASSERT (jerry_value_has_error_flag (res));
```

Comment:
It seems an exclamation mark is missing from here, but was there in the original line.

Reply:
Done.